### PR TITLE
Refactor data validator to use common structure validator; full tests

### PR DESCRIPTION
**Pull request by darinkrauss**
@jh-bate This is the big one. See the `CHANGELOG.md` for a description of the changes.

### CODE REVIEW

File: data/types/basal/test/suppressed.go [hunk 1]
```diff
@@ -1,13 +1,62 @@
...
+	datum := basal.NewSuppressed()
+	datum.Type = pointer.String("basal")
+	datum.DeliveryType = pointer.String("scheduled")
+	datum.Annotations = testData.NewBlobArray()
+	datum.Rate = pointer.Float64(test.RandomFloat64FromRange(basal.RateMinimum, basal.RateMaximum))
+	datum.ScheduleName = pointer.String(NewScheduleName())
+	return datum
+}
+
+func NewSuppressedSuspend() *basal.Suppressed { // Not legal, but used for negative unit tests
```
Comment (jh-bate) [inline data/types/basal/test/suppressed.go]: should it live with the tests then??

File: data/blob_test.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	dataNormalizer "github.com/tidepool-org/platform/data/normalizer"
+	testData "github.com/tidepool-org/platform/data/test"
+	testErrors "github.com/tidepool-org/platform/errors/test"
+	structureValidator "github.com/tidepool-org/platform/structure/validator"
+	"github.com/tidepool-org/platform/test"
+)
+
+var _ = Describe("Blob", func() {
+	Context("ParseBlob", func() {
+		// TODO
```
[the remark this answers is not in the document]
Comment (darinkrauss) [reply data/blob_test.go]: All `Parse*` tests are TODOs in that the parsing code needs to migrate to the `structure.Parser` so no point in writing code that tests the `data.Parser` currently in use.

File: data/types/activity/physical/duration_test.go [hunk 3]
```diff
@@ -1 +1,172 @@
...
+	It("ValueMinimum is expected", func() {
+		Expect(physical.ValueMinimum).To(Equal(0))
+	})
+
+	It("Units returns expected", func() {
+		Expect(physical.Units()).To(Equal([]string{"hours", "minutes", "seconds"}))
+	})
+
+	Context("ParseDuration", func() {
+		// TODO
```
[the remark this answers is not in the document]
Comment (darinkrauss) [reply data/types/activity/physical/duration_test.go]: Parsing code...

File: data/types/activity/physical/physical_test.go [hunk 4]
```diff
@@ -1 +1,205 @@
...
+				Expect(datum.Type).To(Equal("physicalActivity"))
+				Expect(datum.Duration).To(BeNil())
+				Expect(datum.ReportedIntensity).To(BeNil())
+			})
+		})
+	})
+
+	Context("Physical", func() {
+		Context("Parse", func() {
+			// TODO
```
[the remark this answers is not in the document]
Comment (darinkrauss) [reply data/types/activity/physical/physical_test.go]: Parsing code...

File: data/types/basal/test/suppressed.go (hunk 1 above)
Comment (darinkrauss) [reply data/types/basal/test/suppressed.go]: It is with the tests. Any files in any `test` directory is guaranteed to NOT be included in the executable binary, only in the tests.

File: data/blob.go [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+func (b *Blob) Parse(parser ObjectParser) {
+	if obj := parser.Object(); obj != nil {
+		*b = *obj
+	}
+}
+
+func (b *Blob) Validate(validator structure.Validator) {
+	// TODO: Add validation!
```
[the remark this answers is not in the document]
Comment (darinkrauss) [reply data/blob.go]: Yes. I'm not quite sure how to even validate this. A `Blob` is just a `map[string]interface{}` so not sure what it means in this context. The only thing I thought of was that we don't want these huge maps stored in here. The only way I could think of validate that was to marshal it to JSON, but that seems highly inefficient. So, for now, I leave it as a TODO until I think of something. 

Do you have any ideas?

File: data/types/basal/test/suppressed.go (hunk 1 above)
Comment (jh-bate) [reply data/types/basal/test/suppressed.go]: ha ha - I didn't spot that :)

File: data/blob.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (jh-bate) [reply data/blob.go]: as discussed in person no different ideas yet :)

**Pull request merged**